### DISCUSSION TITLE
GitHub container registry - Package not showing on repo

### DISCUSSION THREAD

**Discussion by ghost** [community/community#26266] [Packages]
<p>What’s the logic for packages showing up a the repo level?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/df03c136049159b38b552b5691ca5233ee4f7279.png" data-download-href="/uploads/short-url/vOSwglRWqMpNwjarojdrXBlWR6h.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10137/181096542-d5d11d7d-ef7b-4ec7-9ae2-b9dd995fbccf.png" alt="image" data-base62-sha1="vOSwglRWqMpNwjarojdrXBlWR6h" width="329" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/df03c136049159b38b552b5691ca5233ee4f7279_2_329x500.png, https://aws1.discourse-cdn.com/github/original/2X/d/df03c136049159b38b552b5691ca5233ee4f7279.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/df03c136049159b38b552b5691ca5233ee4f7279.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/df03c136049159b38b552b5691ca5233ee4f7279_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">406×616 20.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I have done the following:</p>
<ul>
<li>Org level switch to GitHub container registry enabled for private images</li>
<li>I have a private Docker image built and pushed to GitHub container registry via GitHub actions</li>
<li>The package shows up at the org level</li>
<li>I <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/connecting-a-repository-to-a-container-image#connecting-a-repository-to-a-user-owned-container-image-on-github" rel="noopener nofollow ugc"> Connected the repository to the container image</a>
</li>
<li>I tried pushing again after they were linked</li>
<li>The repo packages still doesn’t show anything at bottom right of repo</li>
<li>There is no packages tab at the repo level</li>
</ul>
<p>I don’t know that it matters but our org name is Title-Cased but the image repo has to be lower-case.</p>
<p>I also noticed that if I click <code>Publish your first package</code> and then select docker it shows <code>docker.pkg.github.com</code> instead of <code>ghcr.io</code> but maybe that’s just sample docs showing v1 of containers in GitHub packages and not tied into the fact that our org has v2 enabled.</p>

**Comment by noahmatisoff** [comment]
<p>Hi <a class="mention" href="/ghudik-cas">@ghudik-cas</a>!</p>
<p>The team is working on adding container images pushed to GitHub Container Registry to the repository sidebar. We should have them there soon, it’s not an issue with your organization settings. :slight_smile:</p>

**Comment by ghost** [comment]
<p>Ahh, good to to know. Thanks <a class="mention" href="/noahmatisoff">@noahmatisoff</a></p>

**Comment by nolim1t** [comment]
<p>was wondering why this wasn’t working yet it showed up as successfully built</p>

**Comment by clarkbw** [comment, marked as answer]
<p>Please check again, this fix should have rolled out now.</p>

**Comment by ghost** [comment]
<p>I see it now. Thanks!</p>

**Comment by airtower-luna** [comment]
<p>Nice, thank you!</p>
<p>It seem to work only when logged in though, if I open <a href="https://github.com/airtower-luna/hello-ghcr">my test repo</a> in a new browser instance or private window without logging in the “Packages” section say “No packages published”. I can see the package if I click through to packages anyway.</p>

**Comment by clarkbw** [comment]
<p>Good to know. This logged out experience is because of how we roll out feature flags. Once the flag is removed it will become available to everyone. For now it is only for logged in users.</p>

**Comment by airtower-luna** [comment]
<p>Ah, that makes sense. Thanks for the explanation! :slightly_smiling_face:</p>

**Comment by Al-tekreeti** [comment]
<p>It looks still not working. I am not able to see the packages even though the build logs indicate a successful pushing to <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a></p>

**Comment by andyshinn** [comment]
<p>Same. Did this regress? I can push / pull the image. But it is no where in the UI and I cannot manually get to it by name in URL.</p>

**Comment by Al-tekreeti** [comment]
<p>No, still not working</p>

**Comment by justinotero** [comment]
I am just packaging up some JS files.
In Github Actions, it says it's successfully built and created a package. However, cannot find it in the Github UI.

I can pull it down via the `npm` through my CLI.

**Comment by tyrads** [comment]
I can confirm this is not working again with docker container registry.

**Comment by jkovalchik** [comment]
Not showing up for me either. Can successfully push/pull the images to/from GCR but they're not showing up under Packages. Interestingly enough, I _can_ see a couple images built a month ago and one built 9 days ago but nothing more recent than that.

**Comment by Nassiel** [comment]
After all the steps, it still needs to happen for generic packages. Do you have any plans to include this in the future? For instance, I'm pushing an OPA policy in the overall list of packages. It works but is not visible inside the repository despite linking. Thanks @clarkbw :)

**Comment by piotrekkr** [comment]
Same for me. New repo, build and pushed image as `ghcr.io/{repo}:main`. Cannot see in GitHub UI but I can pull...

But... I found how to fix it:

- go to https://github.com/orgs/{NAME}/packages (if you are org), or https://github.com/users/{NAME}/packages (normal user)
- find your manually uploaded package and connect with repository

Now I can see package in repository. But to make full use of new package I also needed to add permissions for repositories so they could read/write.

Lesson is that packages seems not related to repository at first creation. You need to actually attach package to repository and add permissions.

Another lesson: THIS IS REALLY POOR UX. 

I did not even have a link in UI to package list in my repo. I tried attaching /packages to url and then I had page with links to tutorials on how to create package. Finally I managed to view all packages in my organization and then I actually saw my docker image sent hour ago...

// UPDATE 2024-06-03

Did some testing few weeks ago. Seems like if I use Personal Access Token (`PAT`) to create a package, it will not be linked to repository and will need to link it manually. If I use `GITHUB_TOKEN` it will be automatically linked to repository that issues `GITHUB_TOKEN`. Probably better to use `GITHUB_TOKEN` with `packages: write` permissions if it should be assigned to workflow run repository.

// UPDATE 2025-03-12

Forgot to mentions but quite some time ago I found some docs mentioning image labeling as a way to automatically attach package to repository even when using PAT. Like this:
```dockerfile
LABEL org.opencontainers.image.source=https://github.com/octocat/my-repo
LABEL org.opencontainers.image.description="My container image"
LABEL org.opencontainers.image.licenses=MIT
```
Check [pushing docs](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images) and [labeling docs](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#labelling-container-images)

**Reply by thomasgauvin** [reply]
Thanks for providing this @piotrekkr, this resolved my issues

**Reply by victorkt** [reply]
This worked for me as well, I have to agree it's really poor UX

**Comment by NicolasHzGL** [comment]
I'm having the same issue, I'm working in a private repo from an organization. I accidentally published it with a typo, instead of widgets-sdk I published widget-sdk. I fixed it and published it again, the package is being published in GitHub packages though because I can install it in other projects and it doesn't let me publish the same version. The UI is not showing it on the organization packages or the repo packages list.
I can only suspect that my typo linked the package to a repo that doesn't exist inside the organization.

**Reply by NicolasHzGL** [reply]
Hi @clarkbw do you know what could be the issue here?

**Comment by Audiosutras** [comment]
Running into this issue :point_up: check timestamp

**Reply by piotrekkr** [reply]
Hi @Audiosutras. I've updated my [comment](https://github.com/orgs/community/discussions/26266#discussioncomment-6306668), check it maybe it will help. The gist of it is: using PAT will not link package to repo automatically but using `GITHUB_TOKEN` will assign package to repo that issued the `GITHUB_TOKEN`.

**Reply by Audiosutras** [reply]
> Hi @Audiosutras. I've updated my [comment](https://github.com/orgs/community/discussions/26266#discussioncomment-6306668), check it maybe it will help. The gist of it is: using PAT will not link package to repo automatically but using `GITHUB_TOKEN` will assign package to repo that issued the `GITHUB_TOKEN`.

Thanks for responding. Give me 1-3 days to responds.

**Comment by giovannialberto** [comment]
Still not working as of 2024.

This is my [repo](https://github.com/exein-io/kepler). The package is available [here](https://github.com/exein-io/kepler/pkgs/container/kepler) but appears not to be linked to the repository.

**Comment by fcuringa** [comment]
FYI I also had the same issue, doing the following fixed it for my repository:

1. Go to your personal packages page at `https://github.com/{username}?tab=packages` (replace username with your actual username)
2. Find the package in the list, click on it
3. Under "Inherited access", check the box "Inherit access from source repository (recommended)".
4. Click on "package settings" and change its visibility to "public".

After that, the package now shows in my repository page as it should.

**Reply by bnssoftware** [reply]
But what if we want it to be private in a private repository?
